tests/test_config.py

```diff
@@ -313,3 +315,47 @@ def test_compare_default_config_testdata(
         "intentional, run 'make regenerate-testdata' and commit the "
         "updated test data."
     )
+
+
+@pytest.mark.parametrize(
+    "lora_quantize_dtype,additional_args,exception",
+    [
+        ("nf4", {}, None),
+        (None, {}, None),
+        ("nf4", None, None),
+        (None, None, ConfigException),
+        ("invalid", {}, ConfigException),
+        ("nf4", {"lora_alpha": 32}, None),
+    ],
+)
+def test_read_train_profile(
+    lora_quantize_dtype: str | None,
+    additional_args: dict[str, Any] | None,
+    exception: Exception | None,
+    tmp_path_home,
+):
+    # define a profile with yaml
+    profile_data = {
+        "model_path": "/path/to/model",
+        "data_path": "/path/to/data",
+        "ckpt_output_dir": "/path/to/checkpoints",
+        "data_output_dir": "/dev/shm",
+        "max_seq_len": 4096,
+        "max_batch_len": 60_000,
+        "num_epochs": 10,
+        "effective_batch_size": 3840,
+        "save_samples": 250_000,
+        "deepspeed_cpu_ofload_optimizer": True,
+        "lora_rank": 4,
+        "lora_quantize_dtype": lora_quantize_dtype,
+        "is_padding_free": False,
+        "nproc_per_nod": 8,
+        "additional_args": additional_args,
+    }
+    train_profile = pathlib.Path(tmp_path_home, "profile.yaml")
+    with open(train_profile, "w", encoding="utf-8") as outfile:
+        yaml.dump(profile_data, outfile)
+
+    if exception:
+        with pytest.raises(exception):
+            config.read_train_profile(train_profile)
```

Comment:
Did you mean to add this in the amended commit?

Reply:
Yes, these are the tests that validate valid and invalid values for alternating types.